packages/snap-url-manager/src/Translators/Url/UrlTranslator.ts

```diff
@@ -148,13 +148,18 @@ export class UrlTranslator implements Translator {
 			.split('&')
 			.filter((v) => v)
 			.map((kvPair) => {
-				const [key, value] = kvPair.split('=').map((v) => decodeURIComponent(v.replace(/\+/g, ' ')));
-				return { key: key.split('.'), value, type: ParamLocationType.query };
+				try {
+					const [key, value] = kvPair.split('=').map((v) => decodeURIComponent(v.replace(/\+/g, ' ')));
+					return { key: key.split('.'), value, type: ParamLocationType.query };
+				} catch (err) {
+					console.error('Uncaught URIError: URI malformed', kvPair);
+					return { key: ['ss__delete'], value: 'ss__delete', type: ParamLocationType.query };
```

Comment:
Could we just set the value to undefined instead of adding this additional case (since we remove empty params below) ?